prow/cmd/hook/main.go

```diff
@@ -193,41 +194,45 @@ func main() {
 		BugzillaClient:            bugzillaClient,
 	}
 
-	promMetrics := hook.NewMetrics()
+	promMetrics := servemux.NewMetrics()
 
 	defer interrupts.WaitForGracefulShutdown()
 
 	// Expose prometheus metrics
 	metrics.ExposeMetrics("hook", configAgent.Config().PushGateway)
 	pjutil.ServePProf()
 
-	server := &hook.Server{
+	var server servemux.EventHandler
+	server = &hook.Server{
 		ClientAgent:    clientAgent,
 		ConfigAgent:    configAgent,
 		Plugins:        pluginAgent,
 		Metrics:        promMetrics,
 		TokenGenerator: secretAgent.GetTokenGenerator(o.webhookSecretFile),
 	}
+
+	logger := logrus.NewEntry(logrus.New())
+	handler := servemux.NewServeMuxHandler(server, secretAgent.GetTokenGenerator(o.webhookSecretFile), logger)
+
 	interrupts.OnInterrupt(func() {
-		server.GracefulShutdown()
```

Comment:
nit: don't rename a bunch of variables at once, it just makes your diff harder to read